src/main/java/reposense/parser/DateArgumentType.java

```diff
@@ -0,0 +1,28 @@
+package reposense.parser;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Optional;
+
+import net.sourceforge.argparse4j.inf.Argument;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+import net.sourceforge.argparse4j.inf.ArgumentType;
+import reposense.util.Constants;
+
+/**
+ * Verifies and parses a string-formatted date to a {@code Date} object.
+ */
+public class DateArgumentType implements ArgumentType<Optional<Date>> {
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_DATE_STRING_FORMAT = "Invalid Date: %s";
+
+    @Override
+    public Optional<Date> convert(ArgumentParser parser, Argument arg, String value) throws ArgumentParserException {
+        try {
+            return Optional.of(Constants.CLI_ARGS_DATE_FORMAT.parse(value));
+        } catch (ParseException pe) {
```

Comment:
so where does the parse exception come from?

Reply:
Also consider shifting Constants.CLI_ARGS_DATE_FORMAT to here instead.

Reply:
It is actually ```java.text.ParseException```. I will use its full package name to avoid confusion.